test/entities/swap.test.ts

```diff
@@ -0,0 +1,129 @@
+                        ) as `${number}`,
+                        decimals: tokens[1].decimals,
+                        symbol: 'XXX',
+                        name: 'XXX Coin',
+                        index: 1,
+                    },
+                ],
+            }),
+            weightedPoolFactory({
+                id: '0xcb444e90d8198415266c6a2724b7900fb12fc56e000000000000000000000011' as Hex,
+                tokens: [
+                    {
+                        address: tokens[1].address,
+                        weight: '0.5',
+                        balance: String(
+                            parseUnits('100000', tokens[1].decimals),
+                        ) as `${number}`,
+                        decimals: tokens[1].decimals,
+                        symbol: 'XXX',
+                        name: 'XXX Coin',
+                        index: 0,
+                    },
+                    {
+                        address: tokens[2].address,
+                        weight: '0.5',
+                        balance: String(
+                            parseUnits('100000', tokens[2].decimals),
+                        ) as `${number}`,
+                        decimals: tokens[2].decimals,
+                        symbol: 'XXX',
+                        name: 'XXX Coin',
+                        index: 1,
+                    },
+                ],
+            }),
+            weightedPoolFactory({
+                id: '0x177127622c4a00f3d409b75571e12cb3c8973d3c000000000000000000000011' as Hex,
+                tokens: [
+                    {
+                        address: tokens[2].address,
+                        weight: '0.5',
+                        balance: String(
+                            parseUnits('100000', tokens[2].decimals),
+                        ) as `${number}`,
+                        decimals: tokens[2].decimals,
+                        symbol: 'XXX',
+                        name: 'XXX Coin',
+                        index: 0,
+                    },
+                    {
+                        address: tokens[3].address,
+                        weight: '0.5',
+                        balance: String(
+                            parseUnits('100000', tokens[3].decimals),
+                        ) as `${number}`,
+                        decimals: tokens[3].decimals,
+                        symbol: 'XXX',
+                        name: 'XXX Coin',
+                        index: 1,
+                    },
+                ],
+            }),
+        ];
+
+        test('amountIn to be maximally 1% more then expected', () => {
+            const amountOut = TokenAmount.fromRawAmount(tokens[3], 1000000n);
+            const swap = new Swap({
+                paths: [new PathWithAmount(tokens, pools, amountOut)],
+                swapKind: SwapKind.GivenOut,
+            });
+
+            const slippage = 1;
+            const limits = swap.limits(slippage, amountOut);
```

Comment:
I believe you should pass the `expectedAmountIn` as input to the limits calculation, right? 
One way of checking that is that the output should be in 18 decimals as the tokenIn is WETH for this test case.

Reply:
I agree with Bruno about the amount variable name, if the swap kind is "givenOut", the variable name should be "amountIn".

About the test of the 18 decimals, I'm not so sure if this is helpful, because the result can be a fraction with less than 18 decimals

Reply:
It's not just a matter of variable name. We need to get `inputAmount` from swap object, apply slippage to it and check if that is what we expect it to be.